llvm/lib/Transforms/IPO/MergeFunctions.cpp

```diff
@@ -375,9 +375,35 @@ bool MergeFunctions::doFunctionalCheck(std::vector<WeakTrackingVH> &Worklist) {
 }
 #endif
 
+// This function iterates over two functions and compares their
+// metadata to make sure we aren't merging functions which have
+// distinct metadata that is different. We don't care about
+// debug data here, it shouldn't change MIR like CFI would.
+static bool hasDistinctMetadataIntrinsic(const Function &F) {
+  for (const BasicBlock &BB : F) {
+    for (const Instruction &I : BB.instructionsWithoutDebug()) {
+      if (!isa<IntrinsicInst>(&I))
+        continue; // Only intrinsics can reference metadata.
+
+      // Iterate through operands, and return true if you find a distinct
+      // metadata operand.
+      for (unsigned i = 0, e = I.getNumOperands(); i != e;
+           ++i) { // Replaced InstLIt with I
```

Comment:
What's this "Replaced InstLIt with I" comment doing here?

Also why not `for (Value *Op : I.operands())`?